core/clusters/clusters.go

```diff
@@ -680,31 +705,58 @@ func UpdateCluster(name, namespace string, config *ClusterConfig, osclient *ocli
 		return result, generalErr(err, clusterConfigMsg, ErrorCode(err))
 	}
 	workercount := int(finalconfig.WorkerCount)
+	mastercount := int(finalconfig.MasterCount)
 
 	// TODO(tmckay) we need some way to track the current spark config for a cluster,
 	// maybe in annotations. If someone tries to change the spark config for a cluster,
 	// that should be an error at this point (unless we spin all the pods down and
 	// redeploy)
 
 	dcc := osclient.DeploymentConfigs(namespace)
-	dep, err := getDepConfig(dcc, clustername, workerType)
+	err = scaleDep(dcc, clustername, workercount, workerType)
 	if err != nil {
-		return result, generalErr(err, replMsgWorker, ClientOperationCode)
-	} else if dep == nil {
-		return result, generalErr(err, replMsgWorker, ClusterIncompleteCode)
+		return result, err
 	}
-
-	// If the current replica count does not match the request, update the replication controller
-	if int(dep.Spec.Replicas) != workercount {
-		dep.Spec.Replicas = int32(workercount)
-		_, err = dcc.Update(dep)
-		if err != nil {
-			return result, generalErr(err, updateReplMsg, ClientOperationCode)
-		}
+	err = scaleDep(dcc, clustername, mastercount, masterType)
+	if err != nil {
+		return result, err
 	}
 
 	result.Name = name
 	result.Namespace = namespace
 	result.Config = finalconfig
 	return result, nil
 }
+
+
+// Scale a cluster
+// This routine supports a specific scale operation based on immediate values for
+// master and worker counts and does not consider stored configs.
+func ScaleCluster(name, namespace string, masters, workers int, osclient *oclient.Client, client *kclient.Client) error {
```

Comment:
Note to readers,

this routine has been added to core to support the semantics of "scale" in the CLI.  Unlike the general "update" endpoint in oshinko-rest, the CLI scale operation takes only counts and doesn't consider any stored configs.  As such, we need a specific routine to deal with scale from the CLI and it's semantics -- scale workers or masters if a value is specified, otherwise don't.